src/webserver/Dynamic.jl

```diff
@@ -197,4 +196,66 @@ responses[:reshow_cell] = (session::ServerSession, body, notebook::Notebook, cel
     set_output!(cell, run)
     # send to all clients, why not
     putnotebookupdates!(session, notebook, clientupdate_cell_output(notebook, cell))
-end
\ No newline at end of file
+end
+
+responses[:write_file] = (session::ServerSession, body, notebook::Notebook, cell::Cell; initiator::Union{Initiator,Missing}=missing) -> let 
+    path = notebook.path
+    reldir = "$(path |> basename).assets"
+    dir = joinpath(path |> dirname, reldir)
+    file_noext = reduce(*, split(body["name"], ".")[1:end - 1])
+    extension = split(body["name"], ".")[end]
+    save_path = numbered_until_new(joinpath(dir, file_noext); sep=" ", suffix=".$(extension)", create_file=false)
+    
+    if !ispath(dir)
+        mkpath(dir)
+    end
+    success = try
+        io = open(save_path, "w")
+        write(io, body["file"])
+        close(io)
+        true
+    catch e
+        false
+    end
+
+    code = get_template_code(basename(save_path), reldir, body["file"])
+
+    msg = UpdateMessage(:write_file_reply, 
+        Dict(
+            :success => success,
+            :code => code
+        ), notebook, nothing, initiator)
+
+    putclientupdates!(session, initiator, msg)
+end
+
+get_template_code = (filename, directory, iofilecontents) -> begin
+    path = string(raw"$(dirname(@__FILE__))", "/", directory, "/", filename)
```

Comment:
As discussed, please use this format so that it still works after moving / renaming the notebook.

`joinpath(split(@__FILE__, '#')[1] * ".assets", "my_file.jpg")`

Reply:
Note to self: make sure the name change the dir name changes too (maybe that happens already)